core/dbt/cli/flags.py

```diff
@@ -349,6 +353,24 @@ def _assert_mutually_exclusive(
             elif flag_set_by_user:
                 set_flag = flag
 
+    def _validate_event_time_configs(self) -> None:
+        event_time_start = (
+            getattr(self, "EVENT_TIME_START") if hasattr(self, "EVENT_TIME_START") else None
+        )
+        if event_time_start is not None:
+            event_time_end = (
+                getattr(self, "EVENT_TIME_END") if hasattr(self, "EVENT_TIME_END") else None
+            )
+
+            if event_time_end is not None and event_time_start >= event_time_end:
+                raise DbtUsageException(
+                    "Value for `--event-time-start` must be less than `--event-time-end`"
+                )
+            elif event_time_start >= datetime.now():
```

Comment:
These are two separate conditions, so let's do two `if`s?
Also, do we need to specify UTC timezone for `datetime.now()`?

Reply:
To the first part, they are not two separate `if`s. This was intentional as the commit message for ee994c101e610556a398db78b05a08da428632f1 states. We only need to do the second check if `event_time_end` is not set. Said another way, the relation of `event_time_start` to `datetime.now()` is irrelevant if `event_time_end` is set, and thus doesn't need to be checked. 

---

As for the second part, maybe? The issue is that the click parsing of datetimes creates naive datetimes (lacking timezone information). It won't even parse something like `--event-time-start "2024-10-07 13:45:00+00:00"`. It'll give you an error like:

```
Invalid value for '--event-time-start': '2024-10-07 00:00:00+00:00' does not match the formats '%Y-%m-%d', '%Y-%m-%dT%H:%M:%S', '%Y-%m-%d %H:%M:%S'.
```

So our options are to either assume what they give us is UTC, and give the passed in `--event-time-start` a timezone, or compare to a naive datetime. Given the rest of our system assumes UTC, you are right, we should probably do the former instead of what we are currently doing (the latter).